src/Stratis.SmartContracts.CLR.Tests/SmartContracts/SystemContractsDictionary.cs

```diff
@@ -0,0 +1,205 @@
+    }
+
+    public uint GetQuorum(string group)
+    {
+        Assert(!string.IsNullOrEmpty(group));
+        return this.State.GetUInt32($"Quorum:{group}");
+    }
+
+    public void AddSignatory(byte[] signatures, string group, Address address, uint newSize, uint newQuorum)
+    {
+        Assert(!string.IsNullOrEmpty(group));
+        Assert(newSize >= newQuorum, "The number of signatories can't be less than the quorum.");
+
+        Address[] signatories = this.GetSignatories(group);
+        for (int i = 0; i < signatories.Length; i++)
+            Assert(signatories[i] != address, "The signatory already exists.");
+
+        Assert((signatories.Length + 1) == newSize, "The expected size is incorrect.");
+
+        uint nonce = this.State.GetUInt32($"GroupNonce:{group}");
+
+        this.VerifySignatures(signatures, $"{nameof(AddSignatory)}(Nonce:{nonce},Group:{group},Address:{address},NewSize:{newSize},NewQuorum:{newQuorum})");
+
+        System.Array.Resize(ref signatories, signatories.Length + 1);
+        signatories[signatories.Length - 1] = address;
+
+        this.State.SetArray($"Signatories:{group}", signatories);
+        this.State.SetUInt32($"Quorum:{group}", newQuorum);
+        this.State.SetUInt32($"GroupNonce:{group}", nonce + 1);
+    }
+
+    public void RemoveSignatory(byte[] signatures, string group, Address address, uint newSize, uint newQuorum)
+    {
+        Assert(!string.IsNullOrEmpty(group));
+        Assert(newSize >= newQuorum, "The number of signatories can't be less than the quorum.");
+
+        bool found = false;
+        Address[] signatories = this.GetSignatories(group);
+        for (int i = 0; i < signatories.Length; i++)
+        {
+            if (signatories[i] == address)
+            {
+                found = true;
+                for (int j = i + 1; j < signatories.Length; j++)
+                    signatories[j - 1] = signatories[j];
+
+                System.Array.Resize(ref signatories, signatories.Length - 1);
+            }
+        }
+
+        Assert(found, "The signatory does not exist.");
+        Assert(newSize == signatories.Length, "The expected size is incorrect.");
+
+        uint nonce = this.State.GetUInt32($"GroupNonce:{group}");
+
+        this.VerifySignatures(signatures, $"{nameof(RemoveSignatory)}(Nonce:{nonce},Group:{group},Address:{address},NewSize:{newSize},NewQuorum:{newQuorum})");
+
+        this.State.SetArray($"Signatories:{group}", signatories);
+        this.State.SetUInt32($"Quorum:{group}", newQuorum);
+        this.State.SetUInt32($"GroupNonce:{group}", nonce + 1);
+    }
+
+    public bool IsWhiteListed(UInt256 codeHash)
+    {
+        Assert(codeHash != default(UInt256));
+
+        WhiteListEntry whiteListEntry = this.State.GetStruct<WhiteListEntry>(codeHash.ToString());
+
+        return whiteListEntry.CodeHash != default(UInt256);
+    }
+
+    public UInt256 GetCodeHash(string name)
+    {
+        Assert(!string.IsNullOrEmpty(name));
```

Comment:
Imo these asserts are not necessary. If someone send null,empty or none existence key then we return default value already from state.

Reply:
It does make a difference by throwing an error versus simply returning the default. Intuitively throwing an error makes more sense but perhaps there is some reason you don't want to do that for smart contracts. @rowandh , wdyt?

Reply:
As caller of the method, i should to verify that there is hash code or not for given name parameter in any case and i do this checking it is value is default or not. But you added assertion in the case of it is null so i have to verify exception + default value check.

Reply:
This is common way we did in other contracts ussually.

Reply:
If there is insufficient checks something like "Quorum:" could obtain a value, then when "Quorum:" is accessed later a junk value could be returned.